crates/biome_js_analyze/src/lint/nursery/no_nested_ternary.rs

```diff
@@ -0,0 +1,89 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource,
+};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsExpression, JsConditionalExpression};
+use biome_rowan::AstNode;
+
+declare_lint_rule! {
+    /// Disallow nested ternary expressions.
+    ///
+    /// Nesting ternary expressions can make code more difficult to understand.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// const thing = foo ? bar : baz === qux ? quxx : foobar;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// foo ? baz === qux ? quxx() : foobar() : bar();
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// const thing = foo ? bar : foobar;
+    /// ```
+    ///
+    /// ```js
+    /// let thing;
+    ///
+    /// if (foo) {
+    ///     thing = bar;
+    /// } else if (baz === qux) {
+    ///     thing = quxx;
+    /// } else {
+    ///     thing = foobar;
+    /// }
+    /// ```
+    ///
+    pub NoNestedTernary {
+        version: "next",
+        name: "noNestedTernary",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::Eslint("no-nested-ternary")],
+    }
+}
+
+impl Rule for NoNestedTernary {
+    type Query = Ast<JsConditionalExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let alternate = node.alternate().ok()?;
+        let consequent = node.consequent().ok()?;
+        if matches!(alternate, AnyJsExpression::JsConditionalExpression(_))
+            || matches!(consequent, AnyJsExpression::JsConditionalExpression(_))
+        {
+            return Some(());
+        }
```

Comment:
I think this condition would only detect ternaries that are directly nested. We should also flag cases like this:

```ts
x === 5 ? foo(y === 4 ? "foo" : "bar") : "baz"
```

The current condition here also makes it so that we flag the _entire_ parent ternary. Instead, I think it would be more useful to just flag the offending child ternary.

Reply:
> I think this condition would only detect ternaries that are directly nested. We should also flag cases like this:

I'm not sure we should detect non-directly nested ternaries. I feel like there are other edge cases, like:
```
x === 5 ? [y === 4 ? "foo" : "bar"] : "baz"
x === 5 ? {a: y === 4 ? "foo" : "bar"} : "baz"

const result1 = x === 5 ? 'foo' : 'bar';
const result2 = y === 4 ? result1 : 'baz';
```
Handling these cases would definitely complicate the code. Maybe we can add support later if there's enough demand for it.



> The current condition here also makes it so that we flag the entire parent ternary. Instead, I think it would be more useful to just flag the offending child ternary.

Fixed it to now flag only the offending child ternary instead of the entire parent ternary.